src/indieweb_utils/indieauth/relmeauth.py

```diff
@@ -0,0 +1,66 @@
+from typing import List
+from urllib.parse import urlparse as parse_url
+
+import mf2py
+import requests
+from bs4 import BeautifulSoup
+
+from ..utils.urls import canonicalize_url
+
+
+def get_valid_relmeauth_links(url: str) -> List[str]:
+    """
+    Get the valid links on a page that point back to a rel=me URL per RelMeAuth.
+
+    refs: https://indieweb.org/RelMeAuth
+    refs: http://microformats.org/wiki/RelMeAuth
+
+    :url: The url to parse.
+    :type url: str
+    :return: The valid relmeauth links.
+    :rtype: dict
+    """
+    domain = parse_url(url).netloc
+
+    canonical_url = canonicalize_url(url, domain).strip("/")
+
+    mf2_data = mf2py.parse(url=canonical_url)
+
+    rel_me_links = mf2_data["rels"].get("me", [])
+
+    valid_rel_me_links = []
```

Comment:
We could simplify slightly if we make valid_rel_me_links a `set` and just turn it into a list on return.

```suggestion
    valid_rel_me_links: Set[str] = set()
```